### ISSUE TITLE
Unused loss computed during test step in 5.3-torchmetrics

### ISSUE THREAD

**Issue by thomasjpfan** [Lightning-AI/dl-fundamentals#29]
During testing, the loss is computed in `test_step`, but it is not used:

https://github.com/Lightning-AI/dl-fundamentals/blob/66abc069f54fe69769bcebfa091b6feec0e85762/unit05-lightning/5.3-torchmetrics/5.3-part3-torchmetrics-testset.py#L62-L65

One can control `loss` computation as follows:

```python
    def _shared_step(self, batch, compute_loss=True):
        features, true_labels = batch
        logits = self(features)

        loss = F.cross_entropy(logits, true_labels) if compute_loss else None
        predicted_labels = torch.argmax(logits, dim=1)
        return loss, true_labels, predicted_labels

    def test_step(self, batch, batch_idx):
        _, true_labels, predicted_labels = self._shared_step(batch, compute_loss=False)
```

Although, for teaching, it's likely simpler to keep `_shared_step` as is.

**Comment by rasbt** [comment]
Thanks for the note, @thomasjpfan. Yeah, I was thinking about making two different shared step methods but then I decided to keep it simple to avoid visual clutter. 

Maybe  I should have changed

```python
loss, true_labels, predicted_labels = self._shared_step(batch) 
```

to 

```python
_, true_labels, predicted_labels = self._shared_step(batch) 
```

though. Will update this. Thanks!
